core/src/main/java/com/linecorp/armeria/internal/http/Http1ObjectEncoder.java

```diff
@@ -344,9 +343,23 @@ private ChannelFuture doWriteSplitData(
         }
 
         ctx.flush();
+
+        if (data instanceof ByteBufHttpData) {
+            ((ByteBufHttpData) data).buf().release();
```

Comment:
My understanding is

- Need to retain the slice since when written, the channel will release it. If I didn't retain, this release would cause the overall buffer to also be released before getting more slices
- After writing out all the slices, need to release the overall buffer

Let me know if I'm missing something, last time I tried to use retainedSlice I couldn't get it to work so maybe I'm completely misunderstanding it.